src/Services/CoinSelectionService.cs

```diff
@@ -113,13 +115,52 @@ private IBitcoinRequestRepository GetRepository(BitcoinRequestType requestType)
     public async Task LockUTXOs(List<UTXO> selectedUTXOs, IBitcoinRequest bitcoinRequest, BitcoinRequestType requestType)
     {
         // We "lock" the PSBT to the channel operation request by adding to its UTXOs collection for later checking
-        var utxos = selectedUTXOs.Select(x => _mapper.Map<UTXO, FMUTXO>(x)).ToList();
+        var selectedFMUTXOs = selectedUTXOs.Select(x => _mapper.Map<UTXO, FMUTXO>(x)).ToList();
+        var utxos = await _fmutxoRepository.GetByWalletId(bitcoinRequest.Wallet.Id);
 
-        var addUTXOsOperation = await GetRepository(requestType).AddUTXOs(bitcoinRequest, utxos);
+        var utxosToSave = new List<FMUTXO>();
+        var utxosToAdd = new List<FMUTXO>();
```

Comment:
I think we could benefit from some comments here. And also add some TODOs if needed

Reply:
+1 this is hard to read

Reply:
Done 🏁